tests/baseline/cases/staged-simple-directories.yml

```diff
@@ -0,0 +1,46 @@
+# A pushsource library testcase.
+#
+# This file was generated from a template.
+# To regenerate, run test_baseline.py with PUSHSOURCE_UPDATE_BASELINE=1.
+
+# URL of Source to test.
+url: "staged:{{ src_dir }}/tests/staged/data/simple_directories"
+
+# Push items generated from above.
+items:
+- DirectoryPushItem:
+    build: null
+    build_info: null
+    dest:
+    - dest1
+    md5sum: null
+    name: dest1
+    origin: null
+    sha256sum: null
+    signing_key: null
+    src: {{ src_dir }}/tests/staged/data/simple_directories/dest1/RAW
+    state: PENDING
+- DirectoryPushItem:
+    build: null
+    build_info: null
+    dest:
+    - dest2
+    md5sum: null
+    name: dest2
+    origin: null
+    sha256sum: null
+    signing_key: null
+    src: {{ src_dir }}/tests/staged/data/simple_directories/dest2/RAW
```

Comment:
Since there's no special logic in the code to deal with hidden files or .gitkeep files, this test isn't covering an empty RAW directory as was probably intended.  .gitkeep files aren't special so from our code's point of view, it's just like any other file sitting in that directory.

If you really want to cover the empty directory case, you'll probably have to do something special in tests like creating the directory under tmpdir at the time the test runs.  I don't think it matters so much though, as the way the code works now there's negligible risk that it would break for empty directories.